pypeit/utils.py

```diff
@@ -682,50 +713,58 @@ def bspline_profile(xdata, ydata, invvar, profile_basis, inmask = None, upper=5,
     yfit = np.zeros(ydata.shape)
     reduced_chi = 0.
 
-    if invvar.size == 1:
-        outmask = True
-    else:
-        outmask = np.ones(invvar.shape, dtype='bool')
-
-    if inmask is None:
-        inmask = (invvar > 0)
+    # TODO: Instanting these place-holder arrays can be expensive.  Can we avoid doing this?
+    outmask = True if invvar.size == 1 else np.ones(invvar.shape, dtype=bool)
 
-    nin = np.sum(inmask)
-    msgs.info("Fitting npoly =" + "{:3d}".format(npoly) + " profile basis functions, nin=" + "{:3d}".format(nin) + " good pixels")
-    msgs.info("******************************  Iter  Chi^2  # rejected  Rel. fact   ******************************")
-    msgs.info("                              ----  -----  ----------  --------- ")
+    if ingpm is None:
+        ingpm = invvar > 0
 
+    if not quiet:
```

Comment:
Thanks, we badly needed to suppress all this writing to the screen.

Reply:
At the moment, it's still printed.  But we can change the default value of quiet to True.  Let's leave that for another PR, though.